xwiki-platform-core/xwiki-platform-webjars/xwiki-platform-webjars-api/src/main/java/org/xwiki/webjars/internal/filter/LessWebJarsResourceFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.webjars.internal.filter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.io.IOUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.lesscss.compiler.LESSCompiler;
+import org.xwiki.lesscss.compiler.LESSCompilerException;
+import org.xwiki.lesscss.resources.LESSResourceReference;
+import org.xwiki.resource.ResourceReferenceHandlerException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Proceed to the filtering of less resources. The less resources is compiled to css, in the context of the current
+ * skin.
+ *
+ * @version $Id$
+ * @since 13.3RC1
+ */
+@Component
+@Named("less")
+@Singleton
+public class LessWebJarsResourceFilter implements WebJarsResourceFilter
+{
+    @Inject
+    private LESSCompiler lessCompiler;
+
+    @Override
+    public InputStream handle(InputStream resourceStream, String resourceName)
+        throws ResourceReferenceHandlerException
+    {
+        LESSResourceReference lessResourceReference = new WebjarLESSResourceReference(resourceStream, resourceName);
+        try {
+            String compile = this.lessCompiler.compile(lessResourceReference, true, false, false);
+            return IOUtils.toInputStream(compile, UTF_8);
+        } catch (LESSCompilerException e) {
+            throw new ResourceReferenceHandlerException(
+                String.format("Error when compiling the resource [%s]", resourceName), e);
+        }
+    }
+
+    private static class WebjarLESSResourceReference implements LESSResourceReference
```

Comment:
We usually put nested classes at the top. Are you sure there's no Checkstyle violation?

Reply:
No checkstyle violation, and this is not true 100% of the time in XS afaics. But if that helps readability, I'll move it at the top of the class.

Reply:
@tmortagne @vmassol @surli do we have a best practice / code style regarding nested classes? I'm fine to say "if Checkstyle doesn't complain then it's OK" :)

Reply:
I don't remember seeing a nested class not defined at the beginning and my first reflex was also to think that it would fail checkstyle. But I also don't remember ever discussing this.